utils/update_images.py

```diff
@@ -0,0 +1,329 @@
+        tags = [tag.name for tag in repo.get_tags()]
+    except Exception as e:
+        logger.error(f"Error fetching GitHub tags for {repository_path}: {str(e)}")
+    
+    return tags
+
+def detect_images_in_yaml(yaml_data, path=None):
+    """Recursively find image configurations in YAML"""
+    if path is None:
+        path = []
+    
+    found_images = []
+    
+    if isinstance(yaml_data, dict):
+        if "repository" in yaml_data and "tag" in yaml_data:
+            repository = yaml_data["repository"]
+            
+            if not isinstance(repository, str):
+                return found_images
+                
+            if "/" in repository and "." in repository.split("/")[0]:
+                registry = repository.split("/")[0]
+                repo_name = "/".join(repository.split("/")[1:])
+            else:
+                registry = "docker.io"
+                repo_name = repository
+            
+            if repo_name.startswith("localhost") or "<" in repo_name or "example" in repo_name:
+                return found_images
+                
+            name = repository.split("/")[-1].title()
+            if path:
+                name = f"{'.'.join(path)}"
+            
+            current_tag = yaml_data.get("tag", "")
+            version_pattern = None
+            if current_tag:
+                if current_tag.startswith("v") and re.match(r"^v\d+\.\d+\.\d+", current_tag):
+                    version_pattern = r"^v\d+\.\d+\.\d+$"
+                elif re.match(r"^\d+\.\d+\.\d+", current_tag):
+                    version_pattern = r"^\d+\.\d+\.\d+$"
+            
+            found_images.append({
+                "name": name,
+                "registry": registry,
+                "repository": repository,
+                "yaml_path": path,
+                "version_pattern": version_pattern,
+                "current_tag": current_tag
+            })
+        
+        for key, value in yaml_data.items():
+            if isinstance(value, (dict, list)):
+                found_images.extend(detect_images_in_yaml(value, path + [key]))
+    
+    elif isinstance(yaml_data, list):
+        for i, item in enumerate(yaml_data):
+            if isinstance(item, (dict, list)):
+                found_images.extend(detect_images_in_yaml(item, path + [str(i)]))
+    
+    return found_images
+
+def update_yaml_file(file_path, changes):
+    """Update the YAML file with new image tags"""
+    yaml = YAML()
+    yaml.preserve_quotes = True
+    yaml.indent(mapping=2, sequence=4, offset=2)
+    
+    with open(file_path, 'r') as f:
+        data = yaml.load(f)
+    
+    applied_changes = []
+    
+    for change in changes:
+        path = change["path"]
+        new_tag = change["new_tag"]
+        
+        current = data
+        for part in path[:-1]:
+            if part not in current:
+                logger.warning(f"Path {path} not found in YAML")
+                break
```

Comment:
This breaks the loop but continues with the next change, leaving current in an undefined state. You should have `continue` here.